### PR TITLE
Add support for CircleCI

### PR DESCRIPTION
**Pull request by MasoudAsadzade**
* A system from circleci for building OpenSMT and building and testing SMTS with the system default g++ and clang++
*  Enable opensmt in cmake
* comment out readline in cmake
* Termination of SMTS server from client

### CODE REVIEW

File: server/client.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     parser.add_argument('--version', action='version', version=str(version))
     parser.add_argument('host_port', metavar='[host:]port', nargs=1, help='default host=127.0.0.1')
     parser.add_argument('files', metavar='file', nargs='*', help='SMT files to submit. CLI mode if empty')
+    parser.add_argument('-terminate',metavar='q', help='Terminating SMTS Server from client By q')
```
Comment (mmarescotti) [inline server/client.py:20]: The metavar is not used. It'd be nicer to use a flag default false with action `store_true`. See smts.py lines 20-23

File: server/client.py [hunk 2]
```diff
@@ -32,7 +33,10 @@
     try:
         socket = net.Socket()
         socket.connect(address)
-        if args.files:
+        # Terminate server from client
+        if args.terminate is not None and args.terminate[0] == 'q':
+            utils.send_command(args.terminate, socket)
```
Comment (mmarescotti) [inline server/client.py:38]: socket.write can be inlined here

File: server/client.py (hunk 1 above)
Comment (MasoudAsadzade) [reply server/client.py]: Good point! pushed.

File: server/utils.py [hunk 3]
```diff
@@ -125,6 +125,11 @@ def send_file(path, socket):
         'name': name
     }, content)
 
+# Send terminate command
+def send_command(command, socket):
```
[the remark this answers is not in the document]
Comment (MasoudAsadzade) [reply server/utils.py]: Done!

File: server/client.py (hunk 2 above)
Comment (MasoudAsadzade) [reply server/client.py]: Done!

File: src/client/OpenSMT2/CMakeLists.txt [hunk 4]
```diff
@@ -29,4 +29,4 @@ set_target_properties(solver_opensmt PROPERTIES
         COMPILE_FLAGS ${solver_opensmt_flags}
         )
 
-target_link_libraries(solver_opensmt client ${OPENSMT2_LIBRARY} gmp gmpxx z readline)
+target_link_libraries(solver_opensmt client ${OPENSMT2_LIBRARY} gmp gmpxx z ) #readline
```
Comment (aehyvari) [inline src/client/OpenSMT2/CMakeLists.txt:32]: You can just remove readline here for now, maybe forever.

**Review by aehyvari — CHANGES_REQUESTED** [review]
Looks very good!  I left some comments below.  I couldn't figure out how to comment this on the file, but there's an OSX file runtest/DS_Store that should be removed.

File: runtest/run-server-client.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+./server/smts.py -d test.db -o4 -l  &
+sleep 4
+./server/client.py 3000 /home/circleci/externalRepo/opensmt/regression_splitting/instances/init_unsat.smt2;
+./server/client.py 3000 /home/circleci/externalRepo/opensmt/regression_splitting/instances/init_unsat-deep.smt2;
+./server/client.py 3000 /home/circleci/externalRepo/opensmt/regression_splitting/instances/p2-zenonumeric_s6.smt2;
+./server/client.py 3000 /home/circleci/externalRepo/opensmt/regression_splitting/instances/tta_startup_simple_startup_3nodes.synchro.base-deep.smt2;
+./server/client.py 3000  -t
+sleep 3
+
```
[the remark this answers is not in the document]
Comment (MasoudAsadzade) [reply runtest/run-server-client.sh:14]: Done!

File: src/client/OpenSMT2/CMakeLists.txt (hunk 4 above)
Comment (MasoudAsadzade) [reply src/client/OpenSMT2/CMakeLists.txt]: Done!

File: ci/run_travis_smtsCommands.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -ev
+cd build
+cmake -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DCMAKE_CXX_FLAGS="$FLAGS" -DUSE_READLINE:BOOL=${USE_READLINE} -DCMAKE_INSTALL_PREFIX=${INSTALL} ..
+cat CMakeCache.txt
```
[the remark this answers is not in the document]
Comment (MasoudAsadzade) [reply ci/run_travis_smtsCommands.sh]: Done!
./runtest/DS_Store is removed

File: runtest/run-server-client.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (aehyvari) [reply runtest/run-server-client.sh:14]: Was there a reason not to include `${OPENSMTDIR}/regression/QF_UF/php_3_3_40_sat.smt2`?

**Review by aehyvari — APPROVED** [review]
To me this looks good now.

File: runtest/run-server-client.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (MasoudAsadzade) [reply runtest/run-server-client.sh:16]: No, I missed it! Name similarity "php_3_3_40_sat.smt2     vs    php_3_3_40_unsat.smt2 "

**Pull request merged**